core/traits/long_neck.py

```diff
@@ -8,13 +8,23 @@ def __init__(self):
         name = type(self).__name__
         categories = tuple([TraitCategory.EATING])
         super().__init__(name, categories)
-        dispatcher.connect(self.handler_start_feeding_phase, Signal.START_FEEDING_PHASE)
+        dispatcher.connect(
+            self.handler_start_feeding_phase,
+            Signal.BEFORE_FEEDING_PHASE
+        )
 
     def handler_start_feeding_phase(self):
-        # self._owner can be None, otherwise expect it to be a reference.
-        # The reference may resolve to None, so we check for that too.
-        # Otherwise, resolve the reference and call appropriate method.
-        if self._owner_species is not None and self._owner_species() is not None:
+        if self.__owner_species_exists():
             self._owner_species().eat_food()
 
+    def __owner_species_exists(self) -> bool:
```

Comment:
* We have this code duplicated in two places (in the words of Donald Trump... BAD!) We should just implement it on the common parent (as a protected method) and share it between them -- do you agree?
* Also, any reason to delete my comment regarding references etc? I think that point is easy to overlook so would prefer to keep the comment if possible?